packages/vite/src/node/build.ts

```diff
@@ -431,10 +421,12 @@ async function doBuild(
 
   try {
     const buildOutputOptions = (output: OutputOptions = {}): OutputOptions => {
```

Comment:
Even if the input is `.mjs` we still emits `.js`. I am not sure if we should always emit `.mjs` on ESM build, or respect the input (might need to also convert `.mts`)

Reply:
Maybe we should do the same we are doing in v3 for lib mode? https://github.com/vitejs/vite/pull/6827

Reply:
I'll merge this one and work on this logic in another PR.